sql/core/src/main/scala/org/apache/spark/sql/streaming/TestGroupState.scala

```diff
@@ -0,0 +1,161 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.streaming
+
+import org.apache.spark.annotation.{Evolving, Experimental}
+import org.apache.spark.api.java.Optional
+import org.apache.spark.sql.execution.streaming.GroupStateImpl
+import org.apache.spark.sql.execution.streaming.GroupStateImpl._
+
+/**
+ * The extended version of [[GroupState]] interface with extra getters of state machine fields
+ * to improve testability of the [[GroupState]] implementations
+ * which inherit from the extended interface.
+ *
+ * Scala example of using `TestGroupState`:
+ * {{{
+ * import org.apache.spark.sql.streaming.TestGroupState
+ * // other imports
+ *
+ * // test class setups
+ *
+ * test("Structured Streaming state update function") {
+ *   // Creates the prevState input for the state transition function
+ *   // with desired configs. The create() API would guarantee that
+ *   // the generated instance has the same behavior as the one built by
+ *   // engine with the same configs.
+ *   var prevState = TestGroupState.create[UserStatus](
```

Comment:
nit: why is this a var

Reply:
why not be consistent with the docs of GroupState by using the same names and types. like the function name over there is `mappingFunction` and data types are String, Int, and Int?